src/runtime/components/NuxtImg.vue

```diff
@@ -1,18 +1,24 @@
 <template>
   <img
+    v-show="isImageLoaded || !$slots.placeholder"
     ref="imgEl"
-    :class="props.placeholder && !placeholderLoaded ? props.placeholderClass : undefined"
+    :class="props.placeholder && !isImageLoaded ? props.placeholderClass : undefined"
     v-bind="{
       ...isServer ? { onerror: 'this.setAttribute(\'data-error\', 1)' } : {},
       ...imgAttrs,
       ...attrs,
     }"
     :src="src"
   >
+
+  <slot
```

Comment:
my only remaining question is about possible breaking changes.

this makes `<NuxtImg>` into a non-single-root-element component, so (for example) it won't work as the child of a `<Transition>`. (That's why you had to update the tests to add `imageWrapper = wrapper.find('img')`. This would also have to be updated in any project using `NuxtImg`....)

It's also conceptually a change from the idea that `<NuxtImg>` would always just render an `<img>` ...

I'd like to think about whether it's possible to implement this differently - maybe instead, like `<NuxtLink>`, support a `custom` prop that passes down the necessary pieces?

Reply:
I agree with it. I will look into it after Christmas.

Reply:
Should I make another issue or resolve it here?

Reply:
happy for you to resolve it here